pydis_site/apps/content/resources/guides/python-guides/app_commands.md

```diff
@@ -0,0 +1,448 @@
+    def __init__(self) -> None:
+        super().__init__(intents=discord.Intents.default())
+
+    async def setup_hook(self) -> None:
+        ...
+
+'''Another way of creating a "setup_hook" is as follows'''
+
+client = discord.Client(intents=discord.Intents.default())
+async def my_setup_hook() -> None:
+    ...
+
+client.setup_hook = my_setup_hook
+```
+
+# Basic Slash Command application using discord.py.
+
+#### The `CommandTree` class resides within the `app_commands` of discord.py package.
+---
+
+## Slash Command Application with a Client
+
+```python
+import discord
+
+class SlashClient(discord.Client):
+    def __init__(self) -> None:
+        super().__init__(intents=discord.Intents.default())
+        self.tree = discord.app_commands.CommandTree(self)
+
+    async def setup_hook(self) -> None:
+        self.tree.copy_global_to(guild=discord.Object(id=12345678900987654))
+        await self.tree.sync()
+
+client = SlashClient()
+
+@client.tree.command(name="ping", description="...")
+async def _ping(interaction: discord.Interaction) -> None:
+    await interaction.response.send_message("pong")
+
+client.run("token")
+```
+
+
+__**EXPLANATION**__
+
+- `import discord` imports the **discord.py** package.
+- `class SlashClient(discord.Client)` is a class subclassing **Client**. Though there is no particular reason except readability to subclass the **Client** class, using the `Client.setup_hook = my_func` is equally valid.
+- Next up `super().__init__(...)` runs the `__init__` function of the **Client** class, this is equivalent to `discord.Client(...)`. Then, `self.tree = discord.app_commands.CommandTree(self)` creates a CommandTree which acts as the container for slash commaands.
+- Then in the `setup_hook`, `self.tree.copy_global_to(...)` adds the slash command to the guild of which the ID is provided as a `discord.Object` object. Further up, `self.tree.sync()` updates the API with any changes to the Slash Commands.
+- Finishing up with the **Client** subclass, we create an instance of the subclassed Client class which here has been named as `SlashClient` with `client = SlashClient()`.
+- Then using the `command` method of the `CommandTree` we decorate a function with it as `client.tree` is an instance of `CommandTree` for the current application. The command function takes a default argument as said, which acts as the interaction that took place. Catching up is `await interaction.response.send_message("pong")` which sends back a message to the slash command invoker.
+- And the classic old `client.run("token")` is used to connect the client to the discord gateway.
+- Note that the `send_message` is a method of the `InteractionResponse` class and `interaction.response` in this case is an instance of the `InteractionResponse` object. The `send_message` method will not function if the response is not sent within 3 seconds of command invocation. We will discuss how to handle this issue later following the gist.
```

Comment:
Nit: We will discuss how to handle this issue later following the Gist.*

I'm actually not sure anymore, I've seen two utilizations of this, one being "Gist" and the other being "gist".
So I'll leave that to you, I just think they need to stay consistent

Reply:
I'll just have it as `Gist` to keep it consistent.